invenio_communities/views.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2019 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Blueprint definitions."""
+
+from __future__ import absolute_import, print_function
+
+from flask.views import MethodView
+from webargs.flaskparser import use_kwargs
+from webargs import fields, validate
+from invenio_db import db
+
+from flask import Blueprint
+from invenio_communities.api import CommunityMembersCls, MembershipRequestCls
+from flask_security import current_user
+
+def create_blueprint_from_app(app):
+    community_members_rest_blueprint = Blueprint(
+        'invenio_communities',
+        __name__,
+        url_prefix="",
+        template_folder="./templates"
+    )
+    comm_view = CommunityMembersResource.as_view(
+        'community_members_api'
+
+    )
+    community_members_rest_blueprint.add_url_rule(
+        '/communities/<{0}:pid_value>/members'.format(
+            'pid(comid,record_class="invenio_communities.api:Community",'
+            'object_type="com")'),
+        view_func=comm_view,
+    )
+
+    request_view = MembershipRequest.as_view(
+        'community_requests_api'
+    )
+    community_members_rest_blueprint.add_url_rule(
+        '/communities/requests/<token>',
+        view_func=request_view,
+    )
+    return community_members_rest_blueprint
+
+
+class CommunityMembersResource(MethodView):
+    post_args = {
+        'role': fields.Raw(
+            location='json',
+            required=False,
+            validate=[validate.OneOf(['M', 'A', 'C'])]
+        ),# add valid options
+        'email': fields.Email(
+            location='json',
+            required=False
+        ),
+        'invite_type': fields.Raw(
+            location='json',
+            required=True,
+            validate=[validate.OneOf(['invitation', 'request'])]
+        )
+
+    }
+    # TODO: change invite_type name
+    @use_kwargs(post_args)
+    def post(self, email=None, pid_value=None, role=None, invite_type=None):
+        pid, community = pid_value.data
+        if invite_type == 'invitation':
```

Comment:
This can be inferred from if the `current_user` is already part of the community or not:

- `current_user.id in community.members -> "invitation"`
- `current_user.id not in community.members -> "request"`

Reply:
This is not very trustworthy. It's a stretch but imagine the following scenario.

1. I send an email to one of my email addresses.
2. Then I am removed from the community.
3. I then click on the link and accept the invitation to rejoin it.

If I'm not mistaken something like this could then be possible.
I understand that it is a bit complicated currently, so we can also talk IRL for a better solution.